src/formatting.style.md

```diff
@@ -14,6 +14,10 @@ We strive to follow [NEP-1](https://nim-lang.org/docs/nep1.html) for style matte
 func someLongFunctinName(
     alsoLongVariableName: int) = # Double-indent
   discard # back to normal indent
+
+  if someLongCondition and
+      moreLongConditions: # Double-indent
```

Comment:
Most of our code uses aligning in multi-line if statements:

```nim
if someLongCondition and
   moreLongCondtions:
  ...
```

Reply:
This extra level of indent is used quite consistently across large parts codebase and different block-introducing constructs, ie like this `for` as well as proc parameters etc:

https://github.com/status-im/nimbus-eth2/blob/unstable/beacon_chain/spec/state_transition_block.nim#L158=

Ditto for `proc`, `if` etc - the style itself is very common in python and allows visually disentangling the body from the conditions - we had a recent discussion with @Menduist on this subject.

Reply:
For two lines your example is still quite readable, but once you hit a certain amount of lines, it becomes hard to follow what's part of the if, and what's the if body
For instance, this is the if we were talking about:
```nim
          if
              it.connected and
              # avoid negative score peers
              it.score >= 0.0 and
              it notin currentMesh and
              # don't pick explicit peers
              it.peerId notin g.parameters.directPeers and
              # and avoid peers we are backing off
              it.peerId notin backingOff:
            result.add(it)
```

In this case it might actually be cleaner to do
```nim
let cond =
  it.connected and
  it.score >= 0.0 and # etc
if cond:
  result.add(it)
```
But then you create this useless variable

And still, that doesn't solve the issues for procedures:
```nim
proc new*(
  T: type Dialer,
  localPeerId: PeerId,
  connManager: ConnManager,
  transports: seq[Transport],
  ms: MultistreamSelect,
  nameResolver: NameResolver = nil): Dialer =

  T(localPeerId: localPeerId,
    connManager: connManager,
    transports: transports,
    ms: ms,
    nameResolver: nameResolver)
```
Personally I like to put an empty line to signal the start of the body, but that's not ideal either

Reply:
The extra level of indent works well for proc (and avoids the realignment problem that lining things up has):
```
proc new*(
    T: type Dialer,
    localPeerId: PeerId,
    connManager: ConnManager,
    transports: seq[Transport],
    ms: MultistreamSelect,
    nameResolver: NameResolver = nil): Dialer =
  T(...)
```

For `if`, the first condition can usually just sit on the `if` line - this has less churn and makes it easy to tell if the next line is a body (2 spaces) or a continuation of the condition (4 spaces):

```
          if it.connected and
              # avoid negative score peers
              it.score >= 0.0 and
              it notin currentMesh and
              # don't pick explicit peers
              it.peerId notin g.parameters.directPeers and
              # and avoid peers we are backing off
              it.peerId notin backingOff:
            result.add(it)

Reply:
notably, the giant conditions are _quite_ unusual - usually it's more a matter of an extra line or two for `if` conditions and there, it's usually easier to continue them on the next line with an extra indent. like all style decisions, it's hard to make a truly global and universal choice - at some point (once you decide on a style), the code also has to be adapted to that style to look good - that might indeed include an extra variable.

and yes, short-circuit works for a single `let` but doesn't work (for `y) if you do `let a = x, b = y, c = a or b`

Reply:
I've eyeballed a number of the if styles in the code (`if\b[^:]*$`) and it boils down to:

@zah and sometimes @cheatfate line things up -> effectively 3 spaces (emacs default?)

the rest of us do 4-space indent, or variations that end up with 4-space (doing `if  x` with an extra space which causes a both lining up and 4-space.

it's a similar thing as calling a `proc` - emacs users align parameters on a line, but this contradicts the desire to avoid churn on function renames, thus we've chosen to go with 4-space indent on param calls in the style guide.